File: Database/Models/FoodRecord.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FitnessApp.Database.Models
{
    public partial class FoodRecord: ID
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public float Proteins { get; set; }

        [Required]
        public float Carbohydrates { get; set; }

        [Required]
        public float Fats { get; set; }
        [Required]
        public float Calories { get; set; }
        public string? ImageUrl { get; set; }
        [ForeignKey((nameof(UserId)))]
        public User User { get; set; }
        public int UserId { get; set; }
        public DateTime Date { get; set; }

    }
}

File: ClientApp/src/models/FoodRecord.js
export default class FoodRecord {
    id = 0;
    name = '';
    calories = 0;
    proteins = 0.0;
    carbohydrates = 0.0;
    fats = 0.0;
    imageUrl = '';
    user = null;
    date = null;
    

    constructor(id,name,calories,proteins,carbohydrates,fats,imageUrl,user,date){
        this.id = id;
        this.name = name;
        this.calories = calories;
        this.proteins = proteins;
        this.carbohydrates = carbohydrates;
        this.fats = fats;
        this.imageUrl = imageUrl;
        this.user = user;
        this.date = date;
    }
}